### DISCUSSION TITLE
How to handle case of variable names and commands?

### DISCUSSION THREAD

**Discussion by Felienne** [hedyorg/hedy#768] [Large Projects & Plans]
A kid tried this:

```
naam is Hedy
print 'ik heet ' Naam
```

What to do? Do we force all variables names lowercase? Do we catch and give a specific error message? It is not common but it is sensible to expect this to work as if it is the same name.

[Copied over from https://github.com/Felienne/hedy/issues/169]

**Comment by MarleenGilsing** [comment]
I think we should teach them to use lowercase letters, just like we do with the commands. If you can use Name, you should also be able to use Print (and maybe even pRint etc). I expect Python to discriminate upper/lower case letters as well, so then it's a nice and easy fix.

On the other hand, it would be nice if not the variable name but the variable itself doesn't discriminate upper/lowercase letters for the if statements. In the summercamp I've seen some kids who were making programs like: 

name is ask 'What is your name?'
if name is Marleen print 'cool'

And they'd enter 'marleen' and it wouldn't work (or the other way around).

**Reply by Felienne** [reply]
Ow that is a great point, we have to think about commands too and about string values like "marleen" versus "Marleen"!


In our current version Print indeed is not allowed, it will fail, but maybe we should allow it in lower levels for commands and also names, and then later disallow it? We could:
* add a new level for this
* give a special error warning explaining why it does not work (easy for commands and names, harder for string values)
